hal/src/nRF52840/i2c_hal.cpp

```diff
@@ -391,19 +390,20 @@ int32_t HAL_I2C_Request_Data_Ex(HAL_I2C_Interface i2c, const HAL_I2C_Transmissio
         // FIXME: There is a bug in nrfx_twim driver, if we call nrfx_twim_rx repeatedly and quickly,
         // p_cb->busy will be set and never cleared, in this case nrfx_twim_rx always returns busy error
         LOG_DEBUG(TRACE, "BUSY ERROR, restore twi.");
-        twi_uninit(i2c);
```

Comment:
Do we still need to reinitialize I2C here?

Reply:
`HAL_I2C_Reset(i2c, 0, nullptr);` will simulates 9 clocks on SCL followed by calling `HAL_I2C_Begin()`, which will reinitialize I2C.

Reply:
:+1: